### PR TITLE
Make sure Integer can be treated as Float

### LINKED ISSUE #7899: Type error: expected `re` to be Float, but got Integer.
Looks like we are missing conversion from `Integer` to `Float` as following program demonstrates:
```ruby
from Standard.Base import all

type Complex
    Value re:Float im:Float

    new re=0.0 im=0.0 = Complex.Value re im

    + self (that:Complex) = Complex self.re+that.re self.im+that.re

main =
    IO.println <| (Complex.new 1 2) + (Complex.new 3 3)
    IO.println <| (Complex.new 1 5) + 3
```
Every `Integer` value should be autoconverted to `Float` when needed.

### PR DESCRIPTION
**Pull request by JaroslavTulach**
### Pull Request Description

Fix #7899.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] All code follows the guide
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Review by GregoryTravis — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 1]
```diff
@@ -1169,3 +1169,5 @@ type Number_Parse_Error
     to_display_text : Text
     to_display_text self =
         "Could not parse " + self.text.to_text + " as a double."
+
+Float.from (that:Integer) = (1.0 * that):Float
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1174]: Shouldn't we use some builtin that does the conversion instead on relying on semantics of the `*` node?

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1174]: But maybe it doesn't matter too much

**Review by radeusgd — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso (hunk 1 above)
Comment (JaroslavTulach) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1174]: I was searching for some existing function to call. But I found only `floor` & co. No function that would work the opposite way (from `Integer` to `Float`)

Builtin would add fifty lines of code, multiplication by `1.0` is a one-liner. The performance shall be identical.

Comment (radeusgd) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:1174]: Ok, fair

**Pull request merged**